### PR TITLE
Update Milestone process

### PR DESCRIPTION
**Pull request by fryorcraken**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,34 +22,103 @@ The current reporting requirements are 2 folds:
...
+    end
+```
 
-### Epic Owner Responsibilities
+### Accountability
 
-Each epic should have an owner per affected subteam (research, js-waku, go-waku, nwaku).
+Each epic should have an owner per subteam.
+Most epics will have a unique owner (e.g. a Waku Research team member owns a `E:Research` epic).
+For _Dogfood_ and _QA_ epics, one owner per client should be set.
```
Comment (fryorcraken) [inline README.md:110]: Dogfood is a separate epic to ensure it happens. It also encourage interoperability testing so that dogfooding happens across all clients.

We need to try and see if it works.

File: README.md [hunk 2]
```diff
@@ -58,35 +131,50 @@ However, subsequent/other changes may be picked up in parallel or sequentially b
...
 
 The program manager should ensure that epics are getting the right assignee in a timely fashion.
-For example, when research work starts for a given epic, epic owners from development team should be assigned, so they know to participate in discussions.
+For example, when research work starts for a given milestone, epic owners from development team should be assigned, so they know to participate in discussions.
 Program manager should also ensure that issues are being created in a timely fashion,
-an is encouraged to use client PM call as a forum to check epics to be assigned.
-For example, when PoC is near completion then breaking down the work should be started.
+an is encouraged to use client PM call as a forum to check epics to be assigned, for example when a given epic is near completion.
+
+#### Handovers
```
Comment (fryorcraken) [inline README.md:141]: I do not think we need a strict handover process. However, I believe it is worth mentioning.
It helps ensure the quality of the output.
It also means that once a team has handed over, then can close an epic.

Work may happen after a handover, especially in terms of maintenance and support but it is now out of the hand of the team that handed over.

File: README.md [hunk 3]
```diff
@@ -58,35 +131,50 @@ However, subsequent/other changes may be picked up in parallel or sequentially b
...
+
+The following handovers are defined:
+
+| Handover                      | Expectations when handing over                                                                            | Expectations when accepting handover                                                                  |
+|-------------------------------|-----------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------|
+| Research to development teams | - RFC PR is merged <br /> - PoC PR is merged                                                              | - RFC content and PoC are reviewed <br /> - Own code and functionality <br /> - Own minor RFC changes |
```
Comment (fryorcraken) [inline README.md:147]: Regarding RFC, the aim is to ensure that researcher can focus on the next task.
It does not mean that they are totally out of the loop, but that dev engineers are taking ownership and can process with small RFC adjustment raised through implementation and interop.

In case of major issue with a protocol discovered at a late stage, then we can review on an adhoc basis on how to handle it.

File: README.md [hunk 4]
```diff
@@ -58,35 +131,50 @@ However, subsequent/other changes may be picked up in parallel or sequentially b
...
+
+Once the handover is accepted, the given epic can be closed.
 
 ### GitHub Usage
 
 A _Milestone_:
-- MUST have a matching GH issue in the https://github.com/waku-org/pm repo with `milestone` label assigned.
-- MUST have a GH Milestone in https://github.com/waku-org/pm repo, to which relevant _Epics_ are added.
-- SHOULD have a roadmap to delivery done at planning phase, the GH milestone is then used to track progress.
+- MUST have a matching GitHub issue in the https://github.com/waku-org/pm repo with `milestone` label assigned.
```
Comment (fryorcraken) [inline README.md:160]: I believe one of the limitation we have is that a GitHub milestine is only on one repo and hence we need the related epics to be in the same repo.

I don't think there is a way to get the epic issue in the subteam repo.

File: README.md [hunk 5]
```diff
@@ -58,35 +131,50 @@ However, subsequent/other changes may be picked up in parallel or sequentially b
...
+The group or person handing over is expected to initiate a sync (meeting) or async (chat or GitHub) discussion to go through the output and overview.
+
+Once the handover is accepted, the given epic can be closed.
 
 ### GitHub Usage
 
 A _Milestone_:
-- MUST have a matching GH issue in the https://github.com/waku-org/pm repo with `milestone` label assigned.
-- MUST have a GH Milestone in https://github.com/waku-org/pm repo, to which relevant _Epics_ are added.
-- SHOULD have a roadmap to delivery done at planning phase, the GH milestone is then used to track progress.
```
Comment (fryorcraken) [inline README.md:74]: Having a roadmap in the milestone description that is not updated is not great so I removed this and expect to only use github fields for planning. Let's see how it goes.

File: README.md [hunk 6]
```diff
@@ -58,35 +131,50 @@ However, subsequent/other changes may be picked up in parallel or sequentially b
...
+- MUST have assignee(s), who represent the epic owner (see [accountability](#accountability))
 
 A _Task_:
-- MAY be tracked as a todo item in a GH Issue (_Task_ or _Epic_),
+- MAY be tracked as a todo item in a GitHub Issue (_Task_ or _Epic_),
 - OR MAY be tracked as a single GH issue
   - that MUST be labelled with related _Epic_ label (`E:...`),
 - OR MAY be tracked as a GH Pull Request
-    - that MUST be labelled with related _Epic_ label (`E:...`),
+    - that MUST have a reference to the related GitHub _Task_ or _Epic_ issue
```
Comment (fryorcraken) [inline README.md:177]: Nobody label their PR.
I don't think it impacts project management too much so that fine.

But I do expect a reference to the relate issue to appear in a PR so that context is provided.

File: README.md (hunk 1 above)
Comment (chair28980) [reply README.md:127]: Do we envision a new repo dedicated for the dogfood epics?

Comment (fryorcraken) [reply README.md:127]: I think all epic will have to continue to live in pm repo as per https://github.com/waku-org/pm/pull/107#discussion_r1446864392

File: README.md [hunk 7]
```diff
@@ -18,34 +22,120 @@ The current reporting requirements are 2 folds:
...
+| `E:nwaku`    | nwaku                   | MVP quality software                                                               | Bring software to MVP level, proceed with re-architecture of PoC if needed, ensure functionality is usable, refine APIs, auto-generated/API documentation, ensure interoperability works |
+| `E:js-waku`  | js-waku                 | MVP quality software, including all supported env (e.g. React Native)              | Implement protocol in js-waku, same as nwaku.                                                                                                                                            |
+| `E:go-waku`  | go-waku                 | MVP quality software, include all supported bindings (i.e. C and Rust)             | Implement protocol in go-waku. May not always be relevant - need to conclude multiple client discussion. Otherwise, same as nwaku.                                                       |
```
Comment (fryorcraken) [inline README.md:64]: No matter how the multiple client discussion goes. I propose to ensure that Rust and C bindings are considered as support and included in any work

File: README.md [hunk 8]
```diff
@@ -18,34 +22,120 @@ The current reporting requirements are 2 folds:
...
+| `E:Research` | Waku Research           | PoC, RFC, Protocol Simulations/Studies                                             | Initial work done by the research team to create or change a protocol. Engineering-only Milestones may not have such epic                                                                |
+| `E:nwaku`    | nwaku                   | MVP quality software                                                               | Bring software to MVP level, proceed with re-architecture of PoC if needed, ensure functionality is usable, refine APIs, auto-generated/API documentation, ensure interoperability works |
+| `E:js-waku`  | js-waku                 | MVP quality software, including all supported env (e.g. React Native)              | Implement protocol in js-waku, same as nwaku.                                                                                                                                            |
```
Comment (fryorcraken) [inline README.md:63]: react sdk must not fall in the cracks.
Same for React Native once supported.

File: README.md (hunk 7 above)
Comment (fryorcraken) [reply README.md:64]: Should we include this in the same epic than go-waku or have a dedicated one? I 'd suggest same epic @waku-org/go-waku-developers

**Review by jm-clius — COMMENTED** [review]
In general this LGTM. One clarification from my side - this seems to imply that the Epics-per-milestone is _only_ used to distribute the responsibilities towards the same milestone to different subteams. In other words, we won't also use epics to break the milestone into smaller deliverables (that may also span several teams) as I've done in https://www.notion.so/Waku-Research-Milestones-2024-56b1d8c869f14acc934c2d2b2564fa2a ? For example, the Milestone delivery of Store Sync can be broken down into distinct phases: (1) improved Store protocol (2) basic sync protocol. Both of these phases would in turn require effort to be scheduled for each of the subteams. In my mind it make sense to use Epics to also indicate this further breakdown of the milestone.

**Comment by chair28980** [comment]
@jm-clius I agree with your point that Epics should breakdown into appropriately sized batches of work. Using the Milestone you referenced as a rough template of what the work _might_ break down to coming out of the Research phase would be a similar breakdown into Epics as described in the Research doc https://www.notion.so/Waku-Research-Milestones-2024-56b1d8c869f14acc934c2d2b2564fa2a. 

image

```
-  Milestone: Store Upgrade (SU)
- - Epic (SU1) [nwaku]
- - Epic (SU1) [go-waku] -- post mvp approval
- - Epic (SU1) [js-waku] -- post mvp approval
...
- - Epic (SU2) [nwaku]
- - Epic (SU2) [go-waku] -- post mvp approval
- - Epic (SU2) [js-waku] -- post mvp approval
...
```
If the feature moves forward out of the nwaku mvp implementation, the Epics would breakdown by subteams and phases.

File: README.md [hunk 9]
```diff
@@ -18,34 +22,120 @@ The current reporting requirements are 2 folds:
...
+| `E:Dogfood`  | js-waku, go-waku, nwaku | Lab example updates, own nodes updated, etc.                                       | Each dev team proceed by dogfooding the feature/API by using it themselves. Whether it is running their own node, or updating a selected number of examples.                             |
+| `E:Docs`     | Doc                     | Documentation (not auto-generated)                                                 | Document the new feature across all implementations, using the dogfooding output as handover material from engineering teams.                                                            |
+| `E:Eco-Dev`  | Eco Dev                 | Dev Rel assets (examples, video tutorial, etc), comms plan (X threads, blog posts) | Dev Rel can now prepare assets to push the feature to developers, comms can prepare copies to communicate about it, BD can push it to projects and partners.                             |
+
+```mermaid
+flowchart LR
```
Comment (Ivansete-status) [inline README.md:71]: This is awesome :fire: !

File: README.md [hunk 10]
```diff
@@ -18,34 +22,120 @@ The current reporting requirements are 2 folds:
...
+| `E:Dogfood`  | js-waku, go-waku, nwaku | Lab example updates, own nodes updated, etc.                                       | Each dev team proceed by dogfooding the feature/API by using it themselves. Whether it is running their own node, or updating a selected number of examples.                             |
+| `E:Docs`     | Doc                     | Documentation (not auto-generated)                                                 | Document the new feature across all implementations, using the dogfooding output as handover material from engineering teams.                                                            |
+| `E:Eco-Dev`  | Eco Dev                 | Dev Rel assets (examples, video tutorial, etc), comms plan (X threads, blog posts) | Dev Rel can now prepare assets to push the feature to developers, comms can prepare copies to communicate about it, BD can push it to projects and partners.                             |
```
Comment (Ivansete-status) [inline README.md:68]: Shall we restrict to only use low-case in our labels?

```suggestion
| `E:qa`       | Vac/DST                 | RFC-based + functionality based tests, both unit and integration tests.            | Test engineers take over and complete unit tests + add scenarios in integration test framework. In future, also add scenario to benchmark suite.                                         |
| `E:dogfood`  | js-waku, go-waku, nwaku | Lab example updates, own nodes updated, etc.                                       | Each dev team proceed by dogfooding the feature/API by using it themselves. Whether it is running their own node, or updating a selected number of examples.                             |
| `E:docs`     | Doc                     | Documentation (not auto-generated)                                                 | Document the new feature across all implementations, using the dogfooding output as handover material from engineering teams.                                                            |
| `E:eco-dev`  | Eco Dev                 | Dev Rel assets (examples, video tutorial, etc), comms plan (X threads, blog posts) | Dev Rel can now prepare assets to push the feature to developers, comms can prepare copies to communicate about it, BD can push it to projects and partners.                             |
```

File: README.md [hunk 11]
```diff
@@ -18,34 +22,120 @@ The current reporting requirements are 2 folds:
...
-| Task            | Many per Epic                           | Set monthly-ish, delivered weekly      | One subteam or individual                    | Team Member                 | May be one or several piece of work, client specific.                       |  
+Once the team starts building an independent Chat (or other) SDK, the flow will be as above but with research handled by VAC/ACZ and only one dev team:
 
-### Epic Owner Responsibilities
+| Epic Prefix | Owner Sub-team | Output                                             | Description                                                |
+|-------------|----------------|----------------------------------------------------|------------------------------------------------------------|
+| `E:ACZ`     | Vac/ACZ        | RFC                                                | RFC describing a specific, likely agnostic protocol        | 
+| `E:SDK`     | Chat SDK       | PoC and then MVP quality software, Application RFC | Implement the ACZ RFC, define API and application protocol |
```
Comment (Ivansete-status) [inline README.md:119]: What is `ACZ`?
I'm not a big fan of acronyms. imo, better use more descriptive word, i.e. `chat-sdk`.

File: README.md [hunk 12]
```diff
@@ -58,35 +148,50 @@ However, subsequent/other changes may be picked up in parallel or sequentially b
...
+an is encouraged to use client PM call as a forum to check epics to be assigned, for example when a given epic is near completion.
+
+#### Handovers
+
+The following handovers are defined:
+
+| Handover                      | Expectations when handing over                                                                            | Expectations when accepting handover                                                                  |
+|-------------------------------|-----------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------|
+| Research to development teams | - RFC PR is merged <br /> - PoC PR is merged                                                              | - RFC content and PoC are reviewed <br /> - Own code and functionality <br /> - Own minor RFC changes |  
+| Development teams to QA       | - Happy path and selected  error path tests exist <br /> - APIs are implemented to enable interop testing | - Review RFC <br /> - Review existing tests                                                           |
```
Comment (Ivansete-status) [inline README.md:165]: Tiny "typo".

Also, what is considered "enable interop testing"?

```suggestion
| Development teams to QA       | - Happy path and selected error path tests exist <br /> - APIs are implemented to enable interop testing | - Review RFC <br /> - Review existing tests                                                           |
```

File: README.md [hunk 13]
```diff
@@ -58,35 +148,50 @@ However, subsequent/other changes may be picked up in parallel or sequentially b
...
+- MUST have a matching GitHub issue in the https://github.com/waku-org/pm repo with `milestone` label assigned.
+- MUST have a GitHub Milestone in https://github.com/waku-org/pm repo, to which relevant _Epics_ are added.
+- The GitHub milestone MUST be used to track progress.
 
 An _Epic_:
-- MUST have a matching GH issue in the https://github.com/waku-org/pm with `epic` label assigned.
-- MUST have a label with format `E:<epic name>` created across all relevant https://github.com/waku-org/ repos (see [labels.yml](./.github/labels.yml)).
-- SHOULD be added to a GH Milestone.
+- MUST have a matching GitHub issue in the https://github.com/waku-org/pm repo.
+- MUST have a label with format `E:<prefix> <epic name>`.
```
Comment (Ivansete-status) [inline README.md:183]: Should we use dash as a separator instead?
```suggestion
- MUST have a label with format `E:<prefix>-<epic name>`.
```

File: README.md (hunk 6 above)
Comment (Ivansete-status) [reply README.md:194]: IMO, a Task with sub-Tasks is an Epic. Then, I don't see the need for a Task to keep a list of other Tasks

**Review by Ivansete-status — APPROVED** [review]
LGTM! Thanks for it! :100:

File: README.md (hunk 7 above)
Comment (fryorcraken) [reply README.md:64]: I think there is a fair question of scope for each client. Which we are trying to clarify with the multiple client implement.
e.g. if js-waku is only for the browser then relay and service node side of req-res protocol do not need to be implemented.
go-waku scope is currently library for Status so strictly speaking it does not need RLN or autosharding either.

Once we set a scope for a client or binding then it will be clear whether a milestone should be done on such client.

For now, Rust is awkward. Created by Nomos for Nomos but used by TheGraph. TheGraph is valuable user but they do not see intending to use The Waku Network.

Also, our focus was on building The Waku Network, over having an array of language available (see how #42 is planned for later).

However, once we decide that Rust or any specific client should be fully supported, then I think it should be included in milestone work and not be considered a second class citizen.
If we push documentationf or the Rust bindings on docs.waku.org then we should aim for parity, and not wait for a project to tell us "hey this js-waku functionality, would be useful for us in Rust" before we decide to implement.

**Comment by fryorcraken** [comment]
> (1) improved Store protocol (2) basic sync protocol.

Can (1) be done, delivered and useful to developers and users without (2)? If so, then I would consider (1) as a milestone. and (2) as a different milestone.
They can be both in the same track and "Milestone: TWN Store Upgrade (SU)" could simply be viewed as 2024 message reliability track work.

The idea is to have minimal scope for the milestone so we can push them from research to dev rel asap.

In the case of (1) and (2). If I remember correctly as soon as (1) is done in go-waku and nwaku, Status will want to use it. Likely before (2) is finished.
Which means that all the steps after research (nwaku, go-waku, dogfooding, doc) for (1) need to happen independently from (2).

IMO, this also makes it simpler to understand what is done, and what remains to be done.

**Comment by fryorcraken** [comment]
> I agree with your point that Epics should breakdown into appropriately sized batches of work. Using the Milestone you referenced as a rough template of what the work might break down to coming out of the Research phase 

I also accept we need some flexibility for research. Thinking incentivization for example. Research can focus on having a bear minimum PoC but then consider that such PoC would be useless without some addition.

In this case, the "research epic" is not really completed. The work can continue with a second PoC. Once a useful PoC is out, then the scope of the PoC makes the milestone scope and gets pushed across to dev, docs, dev rel etc.

Remaining work can then be scoped in a new milestone.

File: README.md [hunk 14]
```diff
@@ -18,34 +22,120 @@ The current reporting requirements are 2 folds:
...
 #### 2. Monthly Reporting
 
 Monthly reporting is now handled by the Logos insight team.
 
-### Terminology
+### Terminology and Scope
+
+| Name         | Number of             | Timeframe                                      | Team Scope                                      | Owner                  | Description                                           |
+|--------------|-----------------------|------------------------------------------------|-------------------------------------------------|------------------------|-------------------------------------------------------|
+| Deliverable  | 3-5 per year          | Set yearly for grant request                   | Waku Team                                       | Waku Lead              | TBD                                                   |
```
[the remark this answers is not in the document]
Comment (fryorcraken) [reply README.md:35]: _Deliverable_ is something asked by C&J for grant reason.
For now, this can be ignored. I'll actually remove it from the doc for simplicity purposes.

File: README.md [hunk 15]
```diff
@@ -18,34 +22,120 @@ The current reporting requirements are 2 folds:
...
+3. **Transversal:** While the vertical scope of a milestone should be minimal, the delivery should be complete in terms of research, engineering, QA, documentation and dev rel assets so that the feature can be pushed to users once the milestone is marked as complete. Feedback loops should be as small as possible to ensure the value of a milestone is measured in a timely manner.
+4. **Attached Estimate:** An estimate should be associated with the milestone to facilitate the measurement of potential ROI. Additionally, tracking the estimate versus the actual progress is crucial for identifying any deviation and making informed decisions (e.g., deciding whether to continue if we learn the estimate is likely to be overrun).
+
+### Epics and Workflow
+
+A *milestone* is divided in *Epics*. Each *epic* is assigned to a given subteam.
+
+Waku lead is accountable for the delivery of a *milestone*.
```
[the remark this answers is not in the document]
Comment (fryorcraken) [reply README.md:53]: I'll remove it because it's already covered as part of the table in _terminology and scope_. I updated an existing sentence but it is not really useful anymore to clarify it.

File: README.md [hunk 16]
```diff
@@ -18,34 +22,120 @@ The current reporting requirements are 2 folds:
...
+
+Each Waku subteam lead (or selected member) is accountable for the delivery of their epic.
+
+Typically, each *milestone* will be divided in the following *epics*:
+
+| Epic Prefix  | Owner Sub-team          | Output                                                                             | Description                                                                                                                                                                              |
```
[the remark this answers is not in the document]
Comment (fryorcraken) [reply README.md:59]: renamed to "epic label prefix"

File: README.md (hunk 10 above)
Comment (fryorcraken) [reply README.md:68]: I just made client name lowercase for now. I think whatever reads best as these are then used in insight dashboards.
The dashboards are then supposed to be public.

@chair28980 what do you prefer?

File: README.md (hunk 11 above)
Comment (fryorcraken) [reply README.md:119]: The encryption team in Vac. I did not choose their name..

File: README.md (hunk 12 above)
Comment (fryorcraken) [reply README.md:165]: We need to enable QA, ie, Florin, to test this functionality in the interop framework.

So for example, if the API is available on libwaku but there is no way to used it via CLI or REST API then it cannot be tested in the interop framework.

Or maybe, there is some CLI parameter available but it's not practical to be used for interop testing and hence REST API should be updated.

File: README.md (hunk 13 above)
Comment (fryorcraken) [reply README.md:183]: epic name should be in plain english.
We should use spaces so that it read easily. Also keeping it in line with how Status team does it.

For example:

 `E:go-waku Node bandwidth management mechanism epic`

instead of

`E:go-waku-Node bandwidth management mechanism epic`

**Comment by fryorcraken** [comment]
@Ivansete-status (not sure why I could not comment).

> IMO, a Task with sub-Tasks is an Epic. Then, I don't see the need for a Task to keep a list of other Tasks

Similarly to above. The aim is to scope _epics_ to a specific team.
Each team has their own epic for a milestone.
A milestone is minimal in scope.

Now, to achieve your team epic (e.g. `E:nwaku Autosharding for autoscaling`) you need to do a number of tasks.

This framework is very flexible on how your organize the task. Whether it's a bunch of GitHub issues on the repo, or just a check list in the epic issue, or both.

As long as:

- The GitHub issue related to the epic have the right epic label
- The PR related to the epic are referring back to a GitHub issue with the label.

This way nobody has to remember to set epic labels in PRs.

File: README.md (hunk 10 above)
Comment (chair28980) [reply README.md]: @fryorcraken i prefer lowercase :smile:

**Comment by chair28980** [comment]
> > (1) improved Store protocol (2) basic sync protocol.
> 
> Can (1) be done, delivered and useful to developers and users without (2)? If so, then I would consider (1) as a milestone. and (2) as a different milestone. They can be both in the same track and "Milestone: TWN Store Upgrade (SU)" could simply be viewed as 2024 message reliability track work.
> 
> The idea is to have minimal scope for the milestone so we can push them from research to dev rel asap.
> 
> In the case of (1) and (2). If I remember correctly as soon as (1) is done in go-waku and nwaku, Status will want to use it. Likely before (2) is finished. Which means that all the steps after research (nwaku, go-waku, dogfooding, doc) for (1) need to happen independently from (2).
> 
> IMO, this also makes it simpler to understand what is done, and what remains to be done.

I agree with this approach but note that it deviates from the approach of approach of fewer, larger milestones.

**Comment by jm-clius** [comment]
> Can (1) be done, delivered and useful to developers and users without (2)

It can. This is just an example, though. There are other milestones with interdependent "subcomponents" that becomes a bit difficult to schedule as individual milestones with clear user benefits. Take the milestone to provide basic service protocol incentivization: we could divide this into milestones each for Store, Filter, Lightpush, etc. but this now becomes a breakdown that does not quite match the practical reality. We're more likely to complete this as a Store PoC, followed by generalisation to all service protocols and then concerted dogfooding to come up with a blueprint for productionisation. Each of these could be communicated as a milestone in its own right, but the "clear user benefit" IMO is best expressed as the overall goal of "basic service protocol incentivization". Some of these fine-grained milestones may also not even impact any team other than the Research team - PoCs for example can be done up to production plan without any other team involvement. Epics would then allow us to differentiate which tasks, towards the same milestone, belong to which teams. That said, this is mostly a matter of semantics and organisation, so I'm also happy to use finer-grained milestones.

File: README.md (hunk 6 above)
Comment (fryorcraken) [reply README.md:191]: We are abusing the word "epic" to have a common terminology across dashboards.

File: .github/ISSUE_TEMPLATE/milestone.md [hunk 17]
```diff
@@ -21,6 +17,16 @@ assignees: ''
...
 
+# Epics
+
+- research: 
+- js-waku: 
+- nwaku: 
+- go-waku:
+- QA: 
+- Docs:
+- Eco-Dev:
```
[the remark this answers is not in the document]
Comment (fryorcraken) [reply .github/ISSUE_TEMPLATE/milestone.md]: This was meant to have the link tot he various github issues but I opted to remove it as it is dupe info.

**Comment by fryorcraken** [comment]
> > Can (1) be done, delivered and useful to developers and users without (2)
> 
> It can. This is just an example, though. There are other milestones with interdependent "subcomponents" that becomes a bit difficult to schedule as individual milestones with clear user benefits. Take the milestone to provide basic service protocol incentivization: we could divide this into milestones each for Store, Filter, Lightpush, etc. but this now becomes a breakdown that does not quite match the practical reality. We're more likely to complete this as a Store PoC, followed by generalisation to all service protocols and then concerted dogfooding to come up with a blueprint for productionisation. Each of these could be communicated as a milestone in its own right, but the "clear user benefit" IMO is best expressed as the overall goal of "basic service protocol incentivization". Some of these fine-grained milestones may also not even impact any team other than the Research team - PoCs for example can be done up to production plan without any other team involvement. Epics would then allow us to differentiate which tasks, towards the same milestone, belong to which teams. That said, this is mostly a matter of semantics and organisation, so I'm also happy to use finer-grained milestones.

I think it's important to go back to the reason why the new process is designed as it is and I realized I may have not expressed it clearly.

The attributes I am aiming for are:

1. Clear tracking of work across the teams so that when we says that a milestone is delivered, then:
  - it is usable by all types of users (operators, web devs, system devs).
  - It is documented (docs, dev rel)
  - It is of high quality (QA, Dogfooding)
2. Items (epic, milestones) can be easily be closed and marked as complete thanks to:
   - Minimal external dependencies
   - Minimal inter team dependency (e.g. nwaku epic can be closed without waiting for QA)
   - Finite, well-define scope.
3. Each milestone and the effort needed has a clear value thanks to a well-defined, value-driven, minimal, scope.

I understand that you are saying that research may need several PoC before we can handover to dev.

I think it's fine if we have a system where a team may have several epics under a milestone. E.g:

Milestone: Store Incentivization

- `E:Research Store incentivization - Improved protocol`
- `E:Research Store incentivization - Basic sync protocol`\

Then, once All PoC are done and ready to be handed over:

- `E:nwaku Store incentivization`

The point of having a process and some guidelines is to make it easy to operate with the wanted attributes (see above), and hard to go against the attributes.

In the case you mentioned, there are 2 paths:

1. Take store incentivization from PoV to MVP to Prod
2. Take store incentivization to PoV, then take generalization to PoV to MVP to Prod

I would argue that (1) is the preferred path.
Minimal scope to bring a useful value to the users and validate assumptions around req-res protocol incentivization before making it generalized. 
If we decide to increase the scope and go for (2) we need to have a clear justification (e.g risk assessment etc).

**Comment by fryorcraken** [comment]
I am merging to also re-org the files and avoid conflicht with weekly update but we can do more changes down the line. 
It seems the only contention point is around having several epics for a give team (see https://github.com/waku-org/pm/pull/107#issuecomment-1888670559).

**Comment by fryorcraken** [comment]
https://github.com/waku-org/pm/pull/112
